executor/adapter.go

```diff
@@ -498,6 +525,21 @@ func (a *ExecStmt) handlePessimisticLockError(ctx context.Context, err error) (E
 			newForUpdateTS = conflictCommitTS
 		}
 	} else {
+		// this branch update ForUpdateTs and report error to client, do no retry
+		// update ForUpdateTs to prevent async rollback problem
+		// begin; select for update key1(here ErrLocked or other errors(or max_execution_time like util),
+		//        key1 lock not get and async rollback key1 is raised)
+		//        select for update key1 again(this time lock succ(maybe lock released by others))
+		//        the async rollback operation rollbacks the lock just acquired, the commitKey will report error
+		//        txn not found, and before the commit operation other transaction may lock succeed(from the user side,
+		//        the first transaction did not release lock yet)
+		if err != nil {
+			newForUpdateTS, tsErr := a.GetTimestampWithRetry(ctx)
```

Comment:
@coocood Is this appropriate to update `forUpdateTs`  if err != nil (ErrNoWaitSet, ErrLockWaitTimeout ...)

Reply:
yes